service-proxy.js

```diff
@@ -494,6 +492,68 @@ function deletePeerIndex(serviceName, hostPort) {
     deleteIndexEntry(self.knownPeers, hostPort, serviceName);
 };
 
+ServiceDispatchHandler.prototype.ensurePeerConnected =
+function ensurePeerConnected(peer, reason) {
+    var self = this;
+
+    if (peer.isConnected('out')) {
+        return;
+    }
+
+    peer.waitForIdentified(peer.connectTo(), onConnIded);
+
+    function onConnIded(err) {
+        if (err) {
+            self.logger.warn(
+                'failed to ensure outgoing connection to service peer',
+                self.extendLogInfo({
+                    error: err,
+                    peerHostPort: peer.hostPort,
+                    refreshReason: reason
+                }));
+            return;
+        }
+        peer.drain({
+            reason: reason,
```

Comment:
Yup, that's a good educational moment to have anyhow, since we'll drain and decline during graceful shutdown anyhow, which we want them to handle.

Plus partial affinity in #66 will Soon ™ drain ex-affinity peers, so clients just really need to get used to this.